recipes/gtk/4.x/conanfile.py

```diff
@@ -0,0 +1,447 @@
+                "fontconfig::fontconfig",
+            ])
+
+        if self.options.get_safe("with_wayland"):
+            self.cpp_info.components["gtk4-wayland"].set_property("pkg_config_name", "gtk4-wayland")
+            self.cpp_info.components["gtk4-wayland"].requires = ["gtk4"]
+            # https://gitlab.gnome.org/GNOME/gtk/-/blob/4.15.6/gdk/wayland/meson.build#L44-50
+            self.cpp_info.components["gtk4"].requires.extend([
+                "xkbcommon::xkbcommon",
+                "wayland::wayland-client",
+                "wayland::wayland-egl",
+                "wayland-protocols::wayland-protocols",
+            ])
+
+        if is_apple_os(self):
+            self.cpp_info.components["gtk4-macos"].set_property("pkg_config_name", "gtk4-macos")
+            self.cpp_info.components["gtk4-macos"].requires = ["gtk4"]
+            # https://gitlab.gnome.org/GNOME/gtk/-/blob/4.15.6/gdk/macos/meson.build#L46-55
+            self.cpp_info.components["gtk4"].frameworks.extend([
+                "AppKit",
+                "Carbon",
+                "CoreVideo",
+                "CoreServices",
+                "Foundation",
+                "IOSurface",
+                "OpenGL",
+                "QuartzCore",
+            ])
+
+        if self.settings.os == "Windows":
+            self.cpp_info.components["gtk4-win32"].set_property("pkg_config_name", "gtk4-win32")
+            self.cpp_info.components["gtk4-win32"].requires = ["gtk4"]
+            # https://gitlab.gnome.org/GNOME/gtk/-/blob/4.15.6/gdk/win32/meson.build#L60-64
+            self.cpp_info.components["gtk4"].system_libs.extend(["hid", "opengl32"])
+
+        if self.settings.os != "Windows":
+            # https://gitlab.gnome.org/GNOME/gtk/-/blob/4.15.6/meson.build?ref_type=tags#L897-907
+            self.cpp_info.components["gtk4-atspi"].set_property("pkg_config_name", "gtk4-atspi")
+            self.cpp_info.components["gtk4-atspi"].requires = ["gtk4"]
+
+            # https://gitlab.gnome.org/GNOME/gtk/-/blob/4.15.6/meson.build?ref_type=tags#L909-919
+            self.cpp_info.components["gtk4-unix-print"].set_property("pkg_config_name", "gtk4-unix-print")
+            self.cpp_info.components["gtk4-unix-print"].requires = ["gtk4"]
+            self.cpp_info.components["gtk4-unix-print"].includedirs.append(os.path.join("include", "gtk-4.0", "unix-print"))
+
+        if self.options.with_introspection:
+            self.buildenv_info.append_path("GI_GIR_PATH", os.path.join(self.package_folder, "res", "share", "gir-1.0"))
+            self.buildenv_info.append_path("GI_TYPELIB_PATH", os.path.join(self.package_folder, "lib", "girepository-1.0"))
+            self.env_info.GI_GIR_PATH.append(os.path.join(self.package_folder, "res", "share", "gir-1.0"))
+            self.env_info.GI_TYPELIB_PATH.append(os.path.join(self.package_folder, "lib", "girepository-1.0"))
+
+        # TODO: add the following info to all generated .pc files:
+        # targets=wayland x11
```

Comment:
TODO: need to add at least the `targets` value to the .pc file.

libinput fails to build without it (when debug GUI is enabled): https://gitlab.freedesktop.org/libinput/libinput/-/blob/1.25.0/meson.build#L558